cfg/lock.go

```diff
@@ -45,9 +47,46 @@ func (lf *Lockfile) WriteFile(lockpath string) error {
 	return ioutil.WriteFile(lockpath, o, 0666)
 }
 
+// Clone returns a clone of Lockfile
+func (lf *Lockfile) Clone() *Lockfile {
+	n := &Lockfile{}
+	n.Hash = lf.Hash
+	n.Updated = lf.Updated
+	n.Imports = lf.Imports.Clone()
+	n.DevImports = lf.DevImports.Clone()
+
+	return n
+}
+
+// Fingerprint returns a hash of the contents minus the date. This allows for
+// two lockfiles to be compared irrespective of their updated times.
+func (lf *Lockfile) Fingerprint() (string, error) {
```

Comment:
@sdboyer I did some digging and I see what you mean. But, it raised a concern for me.

What would be returned is a `[32]byte`. To compare the byte slices you could use something like `bytes.Equal`. But, that needs a type of `[]byte`. So to handle a comparison you'd end up with...

```go
bytes.Equal(a[:], b[:])
```

The `a[:]` is shorthand for `a[0:len(a)]`. This is needed to convert the type from `[32]byte` to `[]byte`. For anyone using this function this seems non-intuitive.

Do we know why the standard library does things this way?